third_party/winuwp_h264/H264Encoder/H264Encoder.cc

```diff
@@ -204,12 +244,47 @@ int WinUWPH264EncoderImpl::InitWriter() {
   ComPtr<IMFAttributes> encodingAttributes;
   ON_SUCCEEDED(MFCreateAttributes(&encodingAttributes, 1));
 
+  // Set mode explicitly if specified.
+  if (rc_mode_ != RcMode::kUnset) {
+    eAVEncCommonRateControlMode mode;
+    switch (rc_mode_) {
+      case RcMode::kCBR:
+        RTC_LOG_F(LS_INFO) << "Using CBR mode";
+        mode = eAVEncCommonRateControlMode_CBR;
+        break;
+      case RcMode::kVBR:
+        RTC_LOG_F(LS_INFO) << "Using Unconstrained VBR mode";
+        mode = eAVEncCommonRateControlMode_UnconstrainedVBR;
+        break;
+      case RcMode::kQuality:
+        RTC_LOG_F(LS_INFO) << "Using Quality mode";
+        mode = eAVEncCommonRateControlMode_Quality;
+        break;
+      default:
+        RTC_NOTREACHED();
+    }
+
+    ON_SUCCEEDED(encodingAttributes->SetUINT32(
+        CODECAPI_AVEncCommonRateControlMode, mode));
+  }
+
+  // Enable CABAC. This will only have effect in profiles that support it.
+  ON_SUCCEEDED(encodingAttributes->SetUINT32(CODECAPI_AVEncH264CABACEnable,
```

Comment:
Can you log a message for that maybe? Also no idea what CABAC is, so maybe add 1-line explanation why we hard-code enable it?

Reply:
Done